query/src/main/scala/filodb/query/exec/PeriodicSamplesMapper.scala

```diff
@@ -161,9 +161,14 @@ extends Iterator[R] with StrictLogging {
       } catch {
         case e: Exception =>
           val tsReader = LongBinaryVector(nextInfo.getTsVectorAccessor, nextInfo.getTsVectorAddr)
+          val valLen = if (rv.partition.schema.data.columns(rv.valueColID).columnType == ColumnType.DoubleColumn) {
+            val valueReader = DoubleVector(nextInfo.getValueVectorAccessor, nextInfo.getValueVectorAddr)
```

Comment:
Oh, I found a generic version that's better.

rv.partition.schema.data.reader(rv.valueColID, nextInfo.getValueVectorAccessor, nextInfo.getValueVectorAddr)